File: 02_selection_sort/js/selection_sort.js
// Функция для поиска наименьшего элемента массива

function findSmallest(arr) {
    let smallest = arr[0];
    let smallestIndex = 0;
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] < smallest) {
            smallest = arr[i];
            smallestIndex = i;
        }
    }
    return smallestIndex;
}

console.log(findSmallest([5, 3, 6, 2, 10])); 

// ответ 3

// На основе функции выше можно написать функцию сортировки выбором

function selectionSort(arr) {
    let newArr = [];
    const length = arr.length;
    for (let i = 0; i < length; i++) {
        let smallest = findSmallest(arr);
        newArr.push(arr.splice(smallest, 1)[0]);
    }
    return newArr;
}

console.log(selectionSort([5, 3, 6, 2, 10])); 

// [2, 3, 5, 6, 10]

File: 02_selection_sort/python/selection_sort.py
# функция для поиска наименьшего элемента массива

def findSmallest(arr):
    smallest = arr[0]
    smallest_index = 0
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index

print(findSmallest([5, 3, 6, 2, 10]))

# ответ 3


# на основе функции выше можно написать функцию сортировки выбором:

def selectionSort(arr):
    newArr = []
    for i in range(len(arr)):
        smallest = findSmallest(arr)
        newArr.append(arr.pop(smallest))
    return newArr

print(selectionSort([5, 3, 6, 2, 10]))

# ответ [2, 3, 5, 6, 10]
